x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverStep.java

```diff
@@ -39,38 +40,46 @@ public boolean isRetryable() {
     @Override
     public void performAction(IndexMetadata indexMetadata, ClusterState currentClusterState,
                               ClusterStateObserver observer, Listener listener) {
-        boolean indexingComplete = LifecycleSettings.LIFECYCLE_INDEXING_COMPLETE_SETTING.get(indexMetadata.getSettings());
-        if (indexingComplete) {
-            logger.trace(indexMetadata.getIndex() + " has lifecycle complete set, skipping " + RolloverStep.NAME);
-            listener.onResponse(true);
-            return;
-        }
+        IndexAbstraction indexAbstraction = currentClusterState.metadata().getIndicesLookup().get(indexMetadata.getIndex().getName());
```

Comment:
Super minor, but can you add an

```java
assert indexAbstraction != null : "expected the index " + indexName + " to exist in the lookup but it didn't";
```

after this line? I don't think it's going to happen, but we should check it regardless. Optionally, we could make it a real error also (throw an `IllegalStateException`)